### DISCUSSION TITLE
How can I train and test on a dataset with training, validation and testing set in file ?

### DISCUSSION THREAD

**Discussion by dangmanhtruong1995** [open-edge-platform/anomalib#723] [Q&A] [answered]
Hi, I am trying to train a dataset for anomaly detection. My data file structure is basically same as with Keras's ImageDataGenerator: One train, val and test set, each have folders of each class (here is normal and abnormal). However I have noticed that in the example on the main page in github you had this: 

> dataset:
>   name: <name-of-the-dataset>
>   format: folder
>   path: <path/to/folder/dataset>
>   normal_dir: normal # name of the folder containing normal images.
>   abnormal_dir: abnormal # name of the folder containing abnormal images.
>   normal_test_dir: null # name of the folder containing normal test images.
>   task: segmentation # classification or segmentation
>   mask: <path/to/mask/annotations> #optional
>   extensions: null
>   split_ratio: 0.2 # ratio of the normal images that will be used to create a test split
>   image_size: 256
>   train_batch_size: 32
>   test_batch_size: 32
>   num_workers: 8
>   transform_config:
>     train: null
>     val: null
>   create_validation_set: true
>   tiling:
>     apply: false
>     tile_size: null
>     stride: null
>     remove_border_count: 0
>     use_random_tiling: False
>     random_tile_count: 16

I have noticed that there is only normal_test_dir and not abnormal_test_dir. May I ask how can I do this then ? Thank you very much.

**Comment by samet-akcay** [comment, marked as answer]
@dangmanhtruong1995, it is because anomaly tasks do not need `abnormal` images during training, but only `normal` images. In this case, any `abnormal` images will be part of the test directory.

One of the limitations of the `Folder` dataset format is that it is missing `validation` path support. It always creates the validation set by splitting the test set. We could perhaps consider adding option so you could use your own validation set.
